plugin/discovery/find.go

```diff
@@ -96,6 +96,7 @@ func findPluginPaths(kind string, machineName string, dirs []string) []string {
 				continue
 			}
 
+			// FIXME: we pass in GOOS_GOARCH paths directly, so these may not be "legacy"
```

Comment:
I think this is saying that you can say `-plugin-dir=/some/path/linux_amd64`, which would fall into this legacy path. That is true, but I don't know if there's any fixing required here, since all that means is that we'll support both versioned and unversioned plugins in such a directory, whereas we'd only support versioned plugins if it were instead passed as `-plugin-dir=/some/path`.

Perhaps what you meant here is just that "legacy" is the wrong term for this, since it will continue to be supported? In which case I'd agree, but maybe we should just fix this terminology now, since it should just be a change to comments and the log line below, I think?